### PR TITLE
Add config SDK to retrieve config sections as dicts

### PR DESCRIPTION
**Pull request by marijnvanderhorst**
~PR #23 should be merged first.~

## Proposed changes
Adds following functions to query PriceCypher config service.
* `config.index()` to retrieve the available config sections for a dataset
* `config.get_parsed_section()` to retrieve the key-value pairs of a config section as dict.

## Types of changes

What types of changes does your code introduce to this repository?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Unit testing
n/a

### Manual testing
Install snapshot version using the following command.

```bash
pip install pricecypher-sdk==0.5.2rc0.dev0
```

See the updated readme on how to use the newly added functions.

## Further comments
n/a

### CODE REVIEW

File: src/pricecypher/config_sections.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from pricecypher.rest import RestClient
+
+
+class ConfigSections(object):
+    """
+    Config API. Exposes all available operations on dataset configs, like fetching the available sections, and
+    retrieving the key-value pairs of a section and parsing it to a dictionary.
+
+    :param str bearer_token: Bearer token for PriceCypher (logical) API. Needs 'read:configuration' scope.
+    :param str config_base: (optional) Base URL for PriceCypher config service API.
```
Comment (Eline2020) [inline src/pricecypher/config_sections.py:13]: Should dataset_id be included in this docstring?

File: src/pricecypher/config_sections.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        (defaults to the static default_config_base, which by default is https://config.pricecypher.com)
+    :param RestClientOptions rest_options: (optional) Set any additional options for the REST client, e.g. rate-limit.
+        (defaults to None)
+    """
+
+    """ Default user-tool base URL """
+    default_config_base = 'https://config.pricecypher.com'
+
+    def __init__(self, bearer_token, dataset_id, config_base=None, rest_options=None):
```
Comment (Eline2020) [inline src/pricecypher/config_sections.py:22]: Out of curiosity, is type hinting generally not used in `__init__`? (I know the types are already in the docstring above)

File: src/pricecypher/endpoints/config.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        Sections endpoints in config service API.
+        :rtype: SectionsEndpoint
+        """
+        return SectionsEndpoint(self.client, self._url(['api/datasets', self.dataset_id, '/config/sections']))
+
+
+class SectionsEndpoint(BaseEndpoint):
+    """
+    PriceCypher sections endpoints in config service API.
+    """
```
Comment (Eline2020) [inline src/pricecypher/endpoints/config.py:33]: Should the docstring include :param: for init args?

**Review by Eline2020 — COMMENTED** [review]
Code looks good to me! Can't really review the logic, just have a few questions about documentation.

**Review by sonalif — DISMISSED** [review]
Approving, after testing with our DSS and in the DEM AWS env

**Review by levykort1 — DISMISSED** [review]
Approved after Eline comments are resolved

**Review by EmielSchmeink — COMMENTED** [review]
I don't have any comments other than the ones @Eline2020 already gave, the type hinting and docs should be consistent.

File: src/pricecypher/config_sections.py (hunk 2 above)
Comment (marijnvanderhorst) [reply src/pricecypher/config_sections.py:22]: Type hinting is currently not used for any of the function parameters in this project.

File: src/pricecypher/endpoints/config.py (hunk 3 above)
Comment (marijnvanderhorst) [reply src/pricecypher/endpoints/config.py:33]: This class is not "public", it's only used internally by the ConfigEndpoint class. Note that it's also not exported in the `endpoints` package `__init__.py`. 

IMO we don't need to be very strict with private classes/functions. Unless it's really not clear what is going on or e.g. unexpected parameters are required of course. Personally I don't think that's the case here.

File: src/pricecypher/config_sections.py (hunk 1 above)
Comment (marijnvanderhorst) [reply src/pricecypher/config_sections.py:13]: Yea it should, added in commit 365b657.

**Review by Eline2020 — APPROVED** [review]

**Pull request merged**